src/snowflake/snowpark/functions.py

```diff
@@ -7198,10 +7208,18 @@ def udf(
         [Row(MINUS_ONE(10)=9)]
 
     """
-    session = session or snowflake.snowpark.session._get_active_session()
+
+    # Precedence to checking sandbox to avoid any side effects
+    if _is_execution_environment_sandboxed:
+        session = None
+        udf_registration = UDFRegistration(session=session).register
```

Comment:
`udf_registration = UDFRegistration(session=session).register()`?

Reply:
I am assigning the method `register` to the variable `udf_registration`, so `udf_registration` is of type Callable. This is because depending on the session object being None, the registration class is initialized differently. And hence where we are calling the `register` method also changes. I want to hide that complexity behind a simple variable.